### PR TITLE
Categorize HDFS quota exceeded exceptions as user errors and provide useful error messages

### LINKED ISSUE #22462: Trino data lake connectors categorize HDFS quota exceeded exceptions as EXTERNAL but should be USER_ERROR
When using a data lake connector (Hive/Iceberg/etc.) to write data to HDFS using Trino, we may see a [QuotaExceededException](https://hadoop.apache.org/docs/current3/api/org/apache/hadoop/hdfs/protocol/QuotaExceededException.html) (e.g. namespace quota or disk space quota exceeded). This is a userspace issue, but currently we categorize it as an `EXTERNAL` error.

I would like to work on a fix for this, but have a couple of things I'd like to discuss before moving forward:

1. My initial approach would be to modify `HdfsOutputFile` to add a new catch block here for `QuotaExceededException`, which will throw a `TrinoException` with type `USER_ERROR`:
https://github.com/trinodb/trino/blob/6697fe24481a30d37eb91efd62666165acf379c2/lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsOutputFile.java#L102-L105
(I think we would need to modify `HdfsOutputStream#write()` as well to catch disk space quota issues specifically, but need to double-check.)
However I noticed that there is only one place in all of the various `trino-filesystem-*`/`trino-hdfs` modules where we throw a `TrinoException`, so I am wondering if there is a different best-practice for surfacing this kind of issue from the FS layer?

2. Are there analogous concepts on other blob stores (S3/GCS/Azure) that we should handle similarly?

As one example, when writing ORC data from Hive or Iceberg using `OrcFileWriterFactory` ([handled here](https://github.com/trinodb/trino/blob/6697fe24481a30d37eb91efd62666165acf379c2/plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcFileWriterFactory.java#L210)) or `IcebergFileWriterFactory` ([handled here](https://github.com/trinodb/trino/blob/6697fe24481a30d37eb91efd62666165acf379c2/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java#L256)), you get an error with type `(HIVE|ICEBERG)_WRITER_OPEN_ERROR)` and message of simply "Error creating ORC file", which makes it challenging for the end user to understand that there is something on their end to correct (quota issue).

### PR DESCRIPTION
**Pull request by xkrogen**
## Description
Closes #22462. See the issue for context on how this issue manifests and what kind of confusing behavior it can cause.

This PR catches `QuotaExceededException` within the HDFS client, and re-throws them as `TrinoException` with a newly added error code `STORAGE_LIMIT_EXCEEDED` which has category `USER_ERROR`.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(X) Release notes are required, with the following suggested text:

```markdown
# Hive
* Better categorize HDFS quota exceeded exceptions. ({issue}`22962`)

# Iceberg
* Better categorize HDFS quota exceeded exceptions. ({issue}`22962`)

# Hudi
* Better categorize HDFS quota exceeded exceptions. ({issue}`22962`)

# Delta Lake
* Better categorize HDFS quota exceeded exceptions. ({issue}`22962`)
```

### CODE REVIEW

**Comment by xkrogen** [comment]
Bump @electrum @pajaks

**Review by wendigo — APPROVED** [review]

File: core/trino-spi/src/main/java/io/trino/spi/StandardErrorCode.java [hunk 1]
```diff
@@ -153,6 +153,7 @@ public enum StandardErrorCode
     INVALID_PLAN(129, USER_ERROR),
     INVALID_VIEW_PROPERTY(130, USER_ERROR),
     INVALID_ENTITY_KIND(131, USER_ERROR),
+    STORAGE_LIMIT_EXCEEDED(132, USER_ERROR),
```
Comment (wendigo) [inline core/trino-spi/src/main/java/io/trino/spi/StandardErrorCode.java:156]: Just STORAGE_QUOTA_EXCEEDED ?

**Review by electrum — CHANGES_REQUESTED** [review]
My apologies for not following up. We shouldn't throw non-`IOException` from the filesystem. I'll follow up with an alternative.

**Comment by mosabua** [comment]
Given the comment from @electrum I am closing this PR.

**Pull request closed without merging**